Rakefile

```diff
@@ -1,10 +1,121 @@
+# frozen_string_literal: true
+
 require 'rake/extensiontask'
 require 'rspec/core/rake_task'
+require 'rake_compiler_dock'
+
+CLEAN.include FileList['**/*{.o,.so,.dylib,.bundle}'],
+              FileList['**/extconf.h'],
+              FileList['**/Makefile'],
+              FileList['pkg/']
+
+CLOBBER.include FileList['**/tmp'],
+                FileList['**/*.log'],
+                FileList['doc/**'],
+                FileList['tmp/']
+CLOBBER.add("ports/*").exclude(%r{ports/archives$})
+
+RE2_GEM_SPEC = Gem::Specification.load('re2.gemspec')
+
+Gem::PackageTask.new(RE2_GEM_SPEC) do |p|
+  p.need_zip = false
+  p.need_tar = false
+end
 
-Rake::ExtensionTask.new('re2')
+CROSS_RUBY_VERSIONS = %w[3.2.0 3.1.0 3.0.0 2.7.0].join(':')
```

Comment:
Do we really want to support all the way to Ruby 2.3.0? I'm inclined to keep it at 2.7.0 and set the gemspec to:

```ruby
spec.required_ruby_version = ">= 2.7.0", 
```

That way no one can accidentally install a precompiled gem with the wrong Ruby version.

Reply:
If there is no way for us to only ship precompiled binaries for 2.7+ while still supporting system libraries for Rubies older than that then I'm happy to follow @flavorjones' lead with nokogiri and sqlite3 and set the required Ruby for the whole gem to match.